Adds a Heresy phobia, replacing the supernatural phobia on heretic sacrifices

DATA-xPUNGED opened the pull request:
## About The Pull Request

This is a rebase of #72563
This adds a new "Heresy" phobia, which replaces the supernatural phobia on heretic sacrifices. However, sacrificing another heretic does not give them this phobia, but instead removes 3 knowledge points + gives them a different flavour text.
## Why It's Good For The Game

Makes wizards, chaplains, heretics, and cultists able to keep playing the video game as their role when they're sacrificed, while still making the phobia follow it's intended purpose of effectively round removing that person in the heretic's eyes.
## Changelog
:cl:
add: Added a phobia of heresy, which replaces the phobia of the supernatural on heretic sacrifices.
add: Sacrificed heretics won't get this phobia, but instead lose 3 knowledge points.
/:cl:

Helg2 committed:
Yee

If heretic sac's heretic/wizard/cultist, they wont be scared because of themselves.

Helg2 committed:
Yee 1.5

I wonder how to make proper define, so it understands what role_wizard means

Helg2 committed:
Yee 2.0

Helg2 committed:
Yee 3.0

It works

Helg2 committed:
!()

Co-authored-by: Autisem <36102060+Autisem@users.noreply.github.com>

Helg2 committed:
What

Helg2 committed:
what

Helg2 committed:
Herecy

Helg2 committed:
non-existent coma

Helg2 committed:
Forgot those

Helg2 committed:
excessive

Helg2 committed:
fix of something

Helg2 committed:
also those

Helg2 committed:
existent coma

Helg2 committed:
Heresy, not herecy

Helg2 committed:
More to fear

Helg2 committed:
Something like that

But it doesnt even work in build.bat so dunno
fixed some typos i found
i hope it'll show me errors though, because in build its not even linked to the part i added

Helg2 committed:
Merge branch 'master' into supernatural-fears-supernatural

Helg2 committed:
yee

Helg2 committed:
yeet

Helg2 committed:
yee

Helg2 committed:
yee

Helg2 committed:
yee

Helg2 committed:
yee

minor typos change, what am i even doing

Helg2 committed:
yee

Helg2 committed:
phobia of fucking everything

Helg2 committed:
i dunno

Helg2 committed:
yee

Helg2 committed:
Revert "yee"

This reverts commit 92310db6c9df0198ceaabff9a711c5e08ce13bb7.

Helg2 committed:
basic mobing

Helg2 committed:
eh

DATA-xPUNGED committed:
Merge branch 'pr/72563' into Heresy-Phobia

DATA-xPUNGED committed:
Fixes bad rebasing

DATA-xPUNGED committed:
MORE BAD REBASING TO BE FIXED

DATA-xPUNGED committed:
fixes a bad proc

Profakos commented:
Aw dang I never got around to do my version.

Helg2 commented:
Please don't copypaste code from the pr except phobia part, it's just won't work and i'm feeling like a shittone of shame

DATA-xPUNGED commented:
> Please don't copypaste code from the pr except phobia part, it's just won't work and i'm feeling like a shittone of shame

as it is a rebase, i've pretty much copied everything, i am trying to figure out how to MAKE IT work

Helg2 committed:
Merge branch 'Heresy-Phobia' into heresy

Helg2 committed:
yee

Helg2 committed:
didn't mean it

Helg2 committed:
eh

Helg2 committed:
yee

Helg2 committed:
Update traits.dm

Helg2 committed:
lionhunters rifle

DATA-xPUNGED committed:
Merge pull request from Helg2/heresy

Puts more heretic mobs under heretic_summon, heretics explode on sacrifice, Added more stuff to the heresy phobia list

DATA-xPUNGED committed:
Removes un-alphabetized stuff that i wrote

DATA-xPUNGED committed:
Merge remote-tracking branch 'upstream/master' into Heresy-Phobia

DATA-xPUNGED commented:
> Aw dang I never got around to do my version.

If you have idea on how to help feel free to PR it

Helg2 commented on code/_globalvars/phobias.dm:
```diff
@@ -23,6 +24,7 @@ GLOBAL_LIST_INIT(phobia_types, sort_list(list(
 	"spiders",
 	"strangers",
 	"the supernatural",
+	"heresy",
```
Redundant "heresy"

DATA-xPUNGED replied:
I knew i missed one

Helg2 commented on code/_globalvars/phobias.dm:
```diff
@@ -454,6 +488,26 @@ GLOBAL_LIST_INIT(phobia_objs, list(
 		/obj/structure/destructible/eldritch_crucible,
 		/obj/structure/spirit_board,
 	)),
+	"heresy" = typecacheof(list(
+		/obj/effect/floating_blade,
+		/obj/effect/heretic_influence,
+		/obj/effect/heretic_rune,
+		/obj/effect/rune,
+		/obj/effect/visible_heretic_influence,
+		/obj/item/clothing/mask/madness_mask,
+		/obj/item/clothing/neck/heretic_focus,
+		/obj/item/clothing/neck/eldritch_amulet,
+		/obj/item/clothing/suit/hooded/cultrobes/eldritch,
+		/obj/item/codex_cicatrix,
+		/obj/item/reagent_containers/cup/beaker/eldritch,
+		/obj/item/melee/rune_carver,
+		/obj/item/melee/sickly_blade,
+		/obj/item/melee/touch_attack/mansus_fist,
+		/obj/item/toy/eldritch_book,
+		/obj/item/toy/reality_pierce,
+		/obj/structure/destructible/eldritch_crucible,
+		/obj/projectile/curse_hand,
+	)),
```
Redundant and non-alphabetical

Helg2 replied:
```suggestion
```
redundant and non-alphabetical

Helg2 commented on strings/phobia.json:
```diff
@@ -183,7 +183,28 @@
 	],
 
-	"falling": [
+"heresy": [
+		"eldritch",
+		"heretic",
+		"mansus",
+		"Rust",
+		"Flesh",
+		"Ash",
+		"Void",
+		"Blade",
+		"Armsy",
+		"Worm",
+		"Raw Prophet",
+		"Rust Walker",
+		"Stalker",
+		"Ghoul",
+		"Pierced Reality",
+		"Maid in the Mirror",
+		"Hand",
+		"Void Storm"
+	],
+
+"falling": [
```
non alphabetical too

Helg2 commented on code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm:
```diff
@@ -374,7 +376,9 @@
 	sac_target.reagents?.del_reagent(/datum/reagent/inverse/helgrasp/heretic)
 	sac_target.clear_mood_event("shadow_realm")
-	sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/supernatural, TRAUMA_RESILIENCE_MAGIC)
+  
```
Linters are probably failing because of a space here

Helg2 commented on code/_globalvars/phobias.dm:
```diff
@@ -23,6 +24,7 @@ GLOBAL_LIST_INIT(phobia_types, sort_list(list(
 	"spiders",
 	"strangers",
 	"the supernatural",
+	"heresy",
```
```suggestion
```

DATA-xPUNGED committed:
Removes redudant un-alphabetized stuff

DATA-xPUNGED committed:
Unfucking some tabs

DATA-xPUNGED committed:
Removes a space

Co-authored-by: Helg2 <93882977+Helg2@users.noreply.github.com>

DATA-xPUNGED committed:
Readds spaces

DATA-xPUNGED committed:
Merge branch 'Heresy-Phobia' of https://github.com/DATA-xPUNGED/DataStation into Heresy-Phobia

nukularpower commented:
why not have this linked to the hand dodging minigame?   like dodging some number of hands like 80-100% makes it so you dont get the trauma or something like that

DATA-xPUNGED commented:
> why not have this linked to the hand dodging minigame? like dodging some number of hands like 80-100% makes it so you dont get the trauma or something like that

The trauma is an replacement for the old interaction where it just fucking gibbed you on sacrifice. So instead it effectively round removes you in the heretic's eyes. With that in mind i don't think that's a good idea

DATA-xPUNGED committed:
This fixes linters maybe?

The space it is mentioned is just not there anymore, so let's delete the whole line and try again

DATA-xPUNGED committed:
Sacrificed heretics will lose knowledge, and have different flavour text when they come back

DATA-xPUNGED committed:
Uppercase

DATA-xPUNGED committed:
Merge branch 'master' into Heresy-Phobia

DATA-xPUNGED committed:
Merge branch 'master' into Heresy-Phobia

DATA-xPUNGED commented:
I've got no idea why those unit tests are failing tbh, i didn't touch head.dm

DATA-xPUNGED committed:
Merge branch 'master' into Heresy-Phobia

DATA-xPUNGED committed:
Merge remote-tracking branch 'upstream/master' into Heresy-Phobia

DATA-xPUNGED committed:
Merge remote-tracking branch 'upstream/master' into Heresy-Phobia

DATA-xPUNGED committed:
demons are now basicmobs

DATA-xPUNGED commented:
Everyone who i have pestered to try and find out why the unit test isn't working has NO IDEA why it isn't, i give up, I'm manually rebasing this again
